app/src/main/java/com/github/h3lp3rs/h3lp/messaging/RecentMessagesActivity.kt

```diff
@@ -0,0 +1,133 @@
+import java.util.*
+
+
+class RecentMessagesActivity : AppCompatActivity() {
+    private var helpeeId : String? = null
+    private val adapter = GroupAdapter<ViewHolder>()
+
+    // Database where all the conversation ids generated by helpers will be stored
+    private val conversationIdsDb = databaseOf(Databases.CONVERSATION_IDS)
+    private val messagesDatabase = databaseOf(Databases.MESSAGES)
+
+    /**
+     * The map is a synchronizedMap to allow for several conversations to be concurrently added
+     * to it
+     */
+    private var idToItem = Collections.synchronizedMap<String, HelperConversation>(mutableMapOf())
+
+    override fun onCreate(savedInstanceState : Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_latest_messages)
+
+        val bundle = this.intent.extras
+        helpeeId = bundle?.getString(EXTRA_HELPEE_ID) ?: helpeeId
+
+        databaseOf(Databases.MESSAGES).setString("HEWIEWEWE", helpeeId!!)
+
+        adapter.setOnItemClickListener { item, view ->
+            val userItem = item as HelperConversation
+            val intent = Intent(view.context, ChatActivity::class.java)
+            // This is needed to differentiate between sent and received text messages. It will be
+            // compared to the value of Messenger received in a conversation.
+            // If the chat activity was launched from the latest messages activity, we know the user
+            // is a person requesting help
+            intent.putExtra(EXTRA_USER_ROLE, HELPEE)
+            intent.putExtra(EXTRA_CONVERSATION_ID, userItem.getConversationId())
+            startActivity(intent)
+        }
+        recyclerview_latest_messages.adapter = adapter
+
+        listenForHelpers()
+        listenForRemovedHelpers()
+    }
+
+    private fun listenForHelpers() {
+        /**
+         * Adds the corresponding conversation to the view when a helper sends a unique conversation
+         * id to the helpee
+         * @param conversationId The corresponding conversation id
+         */
+        fun onChildAdded(conversationId : String) {
+            val helper = HelperConversation(conversationId)
+            adapter.add(helper)
+            idToItem[conversationId] = helper
+
+        }
+        // Reference to the database of the conversation ids send by the helpers who agreed to
+        // provide help
+
+        conversationIdsDb.addEventListener(helpeeId,
+            String::class.java,
+            { id -> run { onChildAdded(id) } },
+            {})
+    }
+
+    /**
+     * Displays a message using snackbar
+     * @param message message to display
+     */
+    private fun displayMessage(message : String) {
+        Toast.makeText(
+            applicationContext, message, Toast.LENGTH_LONG
+        ).show()
+    }
+
+    private fun listenForRemovedHelpers() {
+        // Event listener that handles deleting the conversation from the view upon deletion
```

Comment:
Just for consistency, maybe you can use the same documentation style as the onChildAdded method above with the @param field